src/java/org/apache/cassandra/db/virtual/AccordDebugKeyspace.java

```diff
@@ -577,6 +603,354 @@ public DataSet data()
+                        "  txn_id text,\n" +
+                        "  command_store_id int,\n" +
+                        "  save_status text,\n" +
+                        "  route text,\n" +
+                        "  durability text,\n" +
+                        "  execute_at text,\n" +
+                        "  txn text,\n" +
+                        "  deps text,\n" +
+                        "  waiting_on text,\n" +
+                        "  writes text,\n" +
+                        "  result text,\n" +
+                        "  participants_owns text,\n" +
+                        "  participants_touches text,\n" +
+                        "  participants_has_touched text,\n" +
+                        "  participants_executes text,\n" +
+                        "  participants_waits_on text,\n" +
+                        "  PRIMARY KEY (txn_id, command_store_id)" +
+                        ')', UTF8Type.instance));
+        }
+
+        @Override
+        public DataSet data()
+        {
+            return this;
+        }
+
+        @Override
+        public boolean isEmpty()
+        {
+            return false;
+        }
+
+        @Override
+        public Partition getPartition(DecoratedKey partitionKey)
+        {
+            String txnIdStr = UTF8Type.instance.compose(partitionKey.getKey());
+            TxnId txnId = TxnId.parse(txnIdStr);
+
+            List<Entry> commands = new ArrayList<>();
+            AccordService.instance().node().commandStores().forEachCommandStore(store -> {
+                Command command = ((AccordCommandStore)store).loadCommand(txnId);
+                if (command != null)
+                    commands.add(new Entry(store.id(), command));
+            });
+
+            if (commands.isEmpty())
+                return null;
+
+            SimpleDataSet ds = new SimpleDataSet(metadata);
+            for (Entry e : commands)
+            {
+                Command command = e.command;
+                ds.row(txnIdStr, e.commandStoreId)
+                  .column("save_status", Objects.toString(command.saveStatus()))
+                  .column("route", Objects.toString(command.route()))
+                  .column("participants_owns", toStr(command, StoreParticipants::owns, StoreParticipants::stillOwns))
+                  .column("participants_touches", toStr(command, StoreParticipants::touches, StoreParticipants::stillTouches))
+                  .column("participants_has_touched", Objects.toString(command.participants().hasTouched()))
+                  .column("participants_executes", toStr(command, StoreParticipants::executes, StoreParticipants::stillExecutes))
+                  .column("participants_waits_on", toStr(command, StoreParticipants::waitsOn, StoreParticipants::stillWaitsOn))
+                  .column("durability", Objects.toString(command.durability()))
+                  .column("execute_at", Objects.toString(command.executeAt()))
```

Comment:
Do we want executes_at_least here?

Reply:
It's part of the `WaitingOn` collection here, which is perhaps confusing, but it looks like we don't print it as part of that so yes I will add an additional row.